plugins/janus_sip.c

```diff
@@ -5252,6 +5264,7 @@ void janus_sip_sofia_callback(nua_event_t event, int status, char const *phrase,
 			/* Notify the application */
 			json_t *notify = json_object();
 			json_object_set_new(notify, "sip", json_string("event"));
+			json_object_set_new(notify, "call_id", json_string(sip->sip_call_id->i_id));
```

Comment:
Is a call-ID always guaranteed to be there even for out of dialog SUBSCRIBE transactions? (probably yes, as part of the SIP standard, but as I rarely use SIP I do forget things sometimes!)

Reply:
It is mandatory in all SIP requests, as per [RFC3261](https://datatracker.ietf.org/doc/html/rfc3261)

> A valid SIP request formulated by a UAC MUST, at a minimum, contain
   the following header fields: To, From, CSeq, Call-ID, Max-Forwards,
   and Via; all of these header fields are mandatory in all SIP
   requests

I first thought about adding safe-guards, but discard the idea since header is supposed to be mandatory

Reply:
Yep, thought so, thanks for the clarification :+1: 
I was just wondering of whether Sofia SIP does indeed enforce those ckecks, e.g., if it's possible for it to receive a message without call-ID and make it available at the application level above.

Reply:
That's what I'm trying to confirmed :grinning: by checking the code of sofia-sip-1.12.11.

Reply:
[SIP parser method `sip_complete_message`](https://github.com/freeswitch/sofia-sip/blob/master/libsofia-sip-ua/sip/sip_parser.c#L657) specifically [checks for mandatory headers](https://github.com/freeswitch/sofia-sip/blob/master/libsofia-sip-ua/sip/sip_parser.c#L709)

Reply:
This is also done in method [sip_sanity_check](http://sofia-sip.sourceforge.net/refdocs/sip/sip__util_8h.html#a82c4aab1711baed02335ddee2f36aad9) 

```
sip_sanity_check(sip_t const *sip)
{
  if (!sip ||
      !((sip->sip_request != NULL) ^ (sip->sip_status != NULL)) ||
      !sip->sip_to ||
      !sip->sip_from ||
      !sip->sip_call_id ||
      !sip->sip_cseq ||
      !sip->sip_via ||
      (sip->sip_flags & MSG_FLG_TRUNC))
    return -1;  /* Bad request */
```

Reply:
`sip_sanity_check` is called in both [agent_recv_request](https://github.com/freeswitch/sofia-sip/blob/master/libsofia-sip-ua/nta/nta.c#L2949) & [agent_recv_response](https://github.com/freeswitch/sofia-sip/blob/master/libsofia-sip-ua/nta/nta.c#L3345)

Reply:
Ack, thanks for checking!